logprep/processor/concatenator/rule.py

```diff
@@ -0,0 +1,151 @@
+"""
+This module defines the rule for the concatenator processor which enables combining multiple source
+fields to one target field
+"""
+
+from ruamel.yaml import YAML
+
+from logprep.filter.expression.filter_expression import FilterExpression
+from logprep.processor.base.rule import InvalidRuleDefinitionError, Rule
+
+yaml = YAML(typ="safe", pure=True)
+
+
+class ConcatenatorRuleError(InvalidRuleDefinitionError):
+    """Base class for Concatenator rule related exceptions."""
+
+    def __init__(self, message: str):
+        super().__init__(f"Concatenator rule ({message})")
+
+
+class InvalidConcatenatorRuleDefinition(ConcatenatorRuleError):
+    """Raise if ConcatenatorRule definition invalid."""
+
+    def __init__(self, definition):
+        message = f"The following Concatenator definition is invalid: {definition}"
+        super().__init__(message)
+
+
+class ConcatenatorRule(Rule):
```

Comment:
consider refactoring this with a fancy new style attrs config class in preparation of a possible upcoming rules refactoring like here:
https://github.com/fkie-cad/Logprep/pull/181/files#diff-649964552be616efe14055d32ad8b5186c1053d3d22aba52f37014c5d70dd8cc

this would be less code

Reply:
I revised it. It greatly simplifies the validation process :+1: